rotkehlchen/chain/gnosis/modules/sdai/decoder.py

```diff
@@ -0,0 +1,128 @@
+from typing import TYPE_CHECKING, Any
+from collections.abc import Callable
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value
+from rotkehlchen.constants import ZERO
+from rotkehlchen.chain.gnosis.modules.sdai.constants import (
+    GNOSIS_SDAI_ADDRESS,
+    CPT_SDAI,
+)
+from rotkehlchen.chain.evm.decoding.utils import maybe_reshuffle_events
+from rotkehlchen.chain.ethereum.decoding.constants import GNOSIS_CPT_DETAILS
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.constants.assets import A_SDAI
+from rotkehlchen.types import ChainID
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface, ReloadableCacheDecoderMixin
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.types import ChecksumEvmAddress
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.gnosis.node_inquirer import GnosisInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+SDAI_DEPOSIT = b'\xdc\xbc\x1c\x05$\x0f1\xff:\xd0g\xef\x1e\xe3\\\xe4\x99wbu.:\tR\x84uED\xf4\xc7\t\xd7'
+SDAI_REDEEM = b'\xfb\xdey} \x1ch\x1b\x91\x05e)\x11\x9e\x0b\x02@|{\xb9jJ,u\xc0\x1f\xc9fr2\xc8\xdb'
+
+
+class SdaiDecoder(DecoderInterface, ReloadableCacheDecoderMixin):
+    def __init__(
+            self,
+            gnosis_inquirer: 'GnosisInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=gnosis_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.sdai = A_SDAI.resolve_to_evm_token()
+
+    def decode_sdai_events(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] == SDAI_DEPOSIT:
+            from_address = context.tx_log.topics[1]
+            to_address = context.tx_log.topics[2]
+            owner_address = to_address
+        elif context.tx_log.topics[0] == SDAI_REDEEM:
+            from_address = None
+            to_address = None
+            owner_address = None
+        else:
+            return DEFAULT_DECODING_OUTPUT
+
+        if from_address is None or to_address is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(context.tx_log.data[32:64]),
+            asset=self.sdai,
```

Comment:
you only seem to use the asset to read the amount. You can use `token_normalized_value_decimals` and hardcode the sDAi decimals since it is always sDAI/xDAI/wxDAI the asset here